src/app/data-model-provider/Provider.h

```diff
@@ -105,6 +105,24 @@ class Provider : public ProviderMetadataTree
     virtual std::optional<ActionReturnStatus> Invoke(const InvokeRequest & request, chip::TLV::TLVReader & input_arguments,
                                                      CommandHandler * handler) = 0;
 
+    /// Workaround function to report attribute change.
+    ///
+    /// When this is invoked, the caller is expected to increment the cluster data version, and the attribute path
+    /// should be marked as `dirty` by the data model provider listener so that the reporter can notify the subscriber
+    /// of attribute changes.
+    /// This function should be invoked when attribute managed by attribute access interface is modified but not
+    /// through an actual Write interaction.
+    /// For example, if the LastNetworkingStatus attribute changes because the NetworkCommissioning driver detects a
+    /// network connection status change and calls SetLastNetworkingStatusValue(). The data model provider can recognize
+    /// this change by invoking this function at the point of change.
+    ///
+    /// This is a workaround function as we cannot notify the attribute change to the data model provider. The provider
+    /// should own its data and versions.
+    ///
+    /// TODO: We should remove this function when the AttributeAccessInterface/CommandHandlerInterface is able to report
+    /// the attribute changes.
+    virtual void Temporary_ReportAttributeChanged(const AttributePathParams & path) = 0;
```

Comment:
Should we maybe move this to `ProviderMetadataTree` instead?

My reasoning is that attribute changed seems more metadata-related (i.e. it increases cluster version). This would keep the provider being the action status on top of metadata.

Reply:
agreed, moved it to the ProviderMetadataTree.